Signatures using the Core SDK (#318)
safe-global/safe-docs: pages/sdk-protocol-kit/guides/safe-signatures.md

# Safe Signatures

Understanding and generating signatures can be challenging. In the **Safe{Core} SDK**, we provide a set of utilities that can assist in using signatures with Safe. In this article, we will explain how signatures work and how to generate them using the `@safe-global/protocol-kit` package.

## Setting up the example Safe Account

> Your Safe Account can be configured with various thresholds and different owner types. An owner can be any Ethereum address, such as:

Author: ```suggestion
Your Safe account can be configured with different values for the threshold and different types of owner. An owner can be any Ethereum address, such as:
```

- External Owned Account (EOA)

> - An smart contract that implements [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) for signature validation, such as Safe Account

Author: ```suggestion
- An smart contract that implements [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) for signature validation, such as another Safe.
```

When the owner is an EOA, an ECDSA signature is generated. When the signer is a smart account, the exact signature algorithm depends on the signer itself.

> In this article, we will use the following Safe Account setup. We've five different Ethereum addresses that can act as signers, namely _owner1_ to _owner5_.

Author: ```suggestion
In this article, we will use the following Safe account setup. We've five different Ethereum addresses that can act as signers, namely `owner1` to `owner5`.
```

- `safeAddress3_4`: 3/4 Safe Account (3 signatures required out of 4 owners)
  - `owner1`
  - `owner2`
  - `signerSafe1_1`: 1/1 Safe Account as child owner
    - `owner3`
  - `signerSafe2_3`: 2/3 Safe Account as another child owner
    - `owner4`
    - `owner5`

All the owners are Ethereum addresses. Here are the addresses for this example:

| Who                | Address                                    |
| ------------------ | ------------------------------------------ |

> | **safeAddress3_4** | 0xb3b3862D8e38a1E965eb350B09f2167B2371D652 |

Author: ```suggestion
| **safe3_4** | 0xb3b3862D8e38a1E965eb350B09f2167B2371D652 |
```

| **owner1**         | 0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1 |
| **owner2**         | 0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0 |
| **signerSafe1_1**  | 0x215033cdE0619D60B7352348F4598316Cc39bC6E |
| **owner3**         | 0x22d491Bde2303f2f43325b2108D26f1eAbA1e32b |
| **signerSafe2_3**  | 0xf75D61D6C27a7CC5788E633c1FC130f0F4a62D33 |
| **owner4**         | 0xE11BA2b4D45Eaed5996Cd0823791E0C93114882d |
| **owner5**         | 0xd03ea8624C8C5987235048901fB614fDcA89b117 |

## Transactions

### Creating the transaction object

We can sign transactions using the `protocol-kit` by creating an instance of the `Safe` class:

```typescript
import Safe from '@safe-global/protocol-kit';

// You can use any compatible adapter, such as Web3Adapter or EthersAdapter
const protocolKit = await Safe.create({
  ethAdapter: ethAdapter1,

>   safeAddress: safeAddress3_4,

Author: ```suggestion
  safeAddress: safe3_4,
```
